pyphare/pyphare_tests/test_pharesee/test_geometry_2d.py

```diff
@@ -0,0 +1,493 @@
+                    {"box": Box([25, 35], [45, 44]), "offset": ([0, 40], [0, 0])},
+                ]
+            },
+        ),
+    )
+    @unpack
+    def test_overlaps(self, refinement_boxes, expected):
+        print("GeometryTest.test_overlaps")
+        test_id = self._testMethodName.split("_")[-1]
+        dim, interp_order, nbr_cells = (2, 1, [20] * 2)
+        hierarchy = self.setup_hierarchy(dim, interp_order, nbr_cells, refinement_boxes, quantities="Bx")
+
+        level_overlaps = hierarchy_overlaps(hierarchy)
+        for ilvl, lvl in enumerate(hierarchy.patch_levels):
+            if ilvl not in expected:
+                continue
+            self.assertEqual(len(expected[ilvl]), len(level_overlaps[ilvl]))
+            for exp, actual in zip(expected[ilvl], level_overlaps[ilvl]):
+                self.assertEqual(actual["box"], exp["box"])
+                self.assertTrue(
+                    (np.asarray(actual["offset"]) == np.asarray(exp["offset"])).all()
+                )
+
+        if 1 in level_overlaps:
+            fig = hierarchy.plot_2d_patches(
+                1, [p.box for p in hierarchy.level(1).patches]
+            )
+            fig.savefig(
+                "hierarchy_2d_lvl_" + str(1) + "_simple_test_" + str(test_id) + ".png"
+            )
+
+
+
+    def test_touch_border(self):
+        print("GeometryTest.test_touch_border")
+        hierarchy = super().basic_hierarchy()
+
+        domain_box = hierarchy.domain_box
+        self.assertFalse(touch_domain_border(Box2D(10, 14), domain_box, "upper"))
+        self.assertFalse(touch_domain_border(Box2D(10, 14), domain_box, "lower"))
+        self.assertTrue(touch_domain_border(Box2D(0, 14), domain_box, "lower"))
+        self.assertTrue(touch_domain_border(Box2D(-5, 14), domain_box, "lower"))
+        self.assertTrue(touch_domain_border(Box2D(-5, 10), domain_box, "lower"))
+        self.assertTrue(touch_domain_border(Box2D(-5, 30), domain_box, "upper"))
+        self.assertTrue(touch_domain_border(Box2D(10, 30), domain_box, "upper"))
+        self.assertTrue(touch_domain_border(Box2D(10, 24), domain_box, "upper"))
+
+
+
+    def test_particle_ghost_area_boxes(self):
+        print("GeometryTest.test_particle_ghost_area_boxes")
+        hierarchy = super().basic_hierarchy()
+
+        expected = {
+            0: [
+                {
+                    "boxes": [
+                        Box([-1, -1], [-1, 10]),
+                        Box([10, -1], [10, 10]),
+                        Box([0, -1], [9, -1]),
+                        Box([0, 10], [9, 10]),
+                    ]
+                },
+                {
+                    "boxes": [
+                        Box([9, 9], [9, 20]),
+                        Box([20, 9], [20, 20]),
+                        Box([10, 9], [19, 9]),
+                        Box([10, 20], [19, 20]),
+                    ]
+                },
```

Comment:
why is there not
``` 
                    "boxes": [
                        Box([-1, 9], [-1, 20]),
                        Box([10, 9], [10, 20]),
                        Box([0, 9], [9, 9]),
                        Box([0, 20], [9, 20]),
                    ]
``` 

and 

``` 
                    "boxes": [
                        Box([9, -1], [9, 10]),
                        Box([20, -1], [20, 10]),
                        Box([10, -1], [19, -1]),
                        Box([10, 10], [19, 19]),
                    ]
``` 

we have 4 patches on L0 I would expect 4*4=16 particle ghost boxes ?

Reply:
the "zip" was hiding the other 8 overlaps inside "actual"